pkg/controller/deployment/deployment_controller.go

```diff
@@ -592,10 +592,6 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	everything := metav1.LabelSelector{}
 	if reflect.DeepEqual(d.Spec.Selector, &everything) {
 		dc.eventRecorder.Eventf(d, v1.EventTypeWarning, "SelectingAll", "This deployment is selecting all pods. A non-empty selector is required.")
-		if d.Status.ObservedGeneration < d.Generation {
```

Comment:
I wouldn't call this unnecessary updates, this gives a clear information that the controller noticed the deployment and reacted on it. It will only update when a deployment will change, so that's not that excessive.